.jenkinsfile

```diff
@@ -17,6 +16,34 @@ properties([
     )
 ])
 
+// ============================================================================
+// Stop current build if superseded by new one.
+// ============================================================================
+
+// Cancels active build if there was a new one.
+@NonCPS
+def cancel_previous_builds()
+{
+    def job_name = env.JOB_NAME
+    def build_number = env.BUILD_NUMBER.toInteger()
+    // Get job name
+    def currentJob = Jenkins.instance.getItemByFullName(job_name)
+
+    // Iterating over the builds for specific job
+    for (def build : currentJob.builds)
+    {
+        // If there is a build that is currently running and it's not current build
+        if (build.isBuilding() && build.number.toInteger() != build_number)
```

Comment:
Does `build.isBuilding()` also include builds with the same id that are in the queue?

Reply:
well, I am not 100% sure. It is a little bit complicated. This script is already executed on the master, but does not have builds spawned yet. So it should actually not have running builds with the same build number. But I am not sure if the master build is also somewhere tracked with the build number and killing would be then never spawn any jobs. So checking against the same build number is a precaution that should avoid such pitfalls.